src/platform/ESP32/MdnsImpl.cpp

```diff
@@ -0,0 +1,241 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "lib/protocols/mdns/platform/Mdns.h"
+
+#include <esp_err.h>
+#include <lwip/ip4_addr.h>
+#include <lwip/ip6_addr.h>
+#include <mdns.h>
+
+#include "platform/CHIPDeviceLayer.h"
+#include "support/CHIPMem.h"
+#include "support/CodeUtils.h"
+#include "support/logging/CHIPLogging.h"
+
+namespace {
+
+static constexpr uint32_t kTimeoutMilli = 3000;
+static constexpr size_t kMaxResults     = 20;
+
+} // namespace
+
+namespace chip {
+namespace Protocols {
+namespace Mdns {
+
+CHIP_ERROR ChipMdnsInit(MdnsAsnycReturnCallback initCallback, MdnsAsnycReturnCallback errorCallback, void * context)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+    esp_err_t espError;
+    uint16_t discriminator;
+    char hostname[11]; // Hostname willbe "CHIP-XXXX"
```

Comment:
This is not the correct hostname format. As stated above, the format is ready from Discovery spec.